File: 프로그래머스/2/87377. 교점에 별 만들기/교점에 별 만들기.js
function calc(eq1,eq2){
    let A = eq1[0]
    let B = eq1[1]
    let E = eq1[2]
    
    let C = eq2[0]
    let D = eq2[1]
    let F = eq2[2]
    
    if(A*D === B*C){
        return
    }
    let X = (B*F - E*D) / (A*D - B*C)
    let Y = (E*C - A*F) / (A*D - B*C)
    
    if(Number.isInteger(X) && Number.isInteger(Y)){
        return [parseInt(X),parseInt(Y)]
    }
}

function solution(line){
    let meets = [];
    // 교점 구하기 
    for(let i=0; i<line.length; i++){
        for(let j=i+1; j<line.length; j++){
            let eq1 = [line[i][0],line[i][1],line[i][2]]
            let eq2 = [line[j][0],line[j][1],line[j][2]]
            // meet의 return은 배열의 형태일것 
            let meet = calc(eq1,eq2)
            if(meet){
                meets.push(meet)
            }
        }
    }
    
    let x_meets = []
    let y_meets = []
    for(meet of meets){
        x_meets.push(meet[0])
        y_meets.push(meet[1])
    }
    
    // 보드판 setting
    let r1 = Math.min(...x_meets)
    let r2 = Math.max(...x_meets) + 1
    let c1 = Math.min(...y_meets)
    let c2 = Math.max(...y_meets) + 1
    
    let locs = Array.from(Array(c2-c1),() => Array(r2-r1).fill('.')) 
    
    for(meet of meets){
        let x = meet[0]
        let y = meet[1]
        locs[y-c1][x-r1] = '*'
    }
    locs.reverse()

    return locs.map((loc) => loc.join(""))
}

File: 프로그래머스/2/87377. 교점에 별 만들기/교점에 별 만들기.py
from itertools import combinations

def calc(eq1,eq2):
    x1,y1,c1 = eq1
    x2,y2,c2 = eq2
    
    # 평행 또는 일치일때 zd error 방지
    if x1*y2 == y1*x2:
        return 
    
    # 교점 
    X = (y1*c2 - c1*y2) / (x1*y2 - y1*x2)
    Y = (c1*x2 - x1*c2) / (x1*y2 - y1*x2)
    
    if X == int(X) and Y == int(Y):
        return [int(X), int(Y)]

def solution(line):
    answer = []
    meets = []
    # 모든 별을 포함하는 최소한의 크기만 나타내면 된다. 
    # 주어진 다항식을 기준으로 교점을 찾아야함 
    
    # 주어진 line에서 2개의 교점을 찾아야 하는데
    # 그를 위해선 조합이 필요하다.
    for eq1,eq2 in combinations(line,2):
        meet = calc(eq1,eq2)
        if meet:
            meets.append(meet)
    
    r1,r2 = min(meets,key=lambda x:x[0])[0], max(meets,key=lambda x:x[0])[0] + 1
    c1,c2 = min(meets,key=lambda x:x[1])[1], max(meets,key=lambda x:x[1])[1] + 1
    locs = [['.']*(r2-r1) for _ in range(c2-c1)]
    
    for x,y in meets:
        locs[y-c1][x-r1] = '*'
    locs.reverse()
    
    answer = [''.join(loc) for loc in locs]
    
    return answer